docs/workflow.md

```diff
@@ -0,0 +1,122 @@
+# GitHub Workflow
+
+Our GitHub workflow is a so called triangular workflow:
+
+![Github Worflow](./img/git-workflow.svg)
+
+This [main repository](https://github.com/multigres/multigres) is called _upstream_.
+
+You develop and commit your changes in a clone of our upstream repository (shown as _local_ in the image above).
+Then you push your changes to your forked repository (_origin_) and send us a pull request.
+Eventually, we will merge your pull request back into the _upstream_ repository.
+
+## Remotes
+
+Since you should have cloned the repository from your fork, the `origin` remote should look like this:
+
+```bash
+$ git remote -v
+origin  git@github.com:<yourname>/multigres.git (fetch)
+origin  git@github.com:<yourname>/multigres.git (push)
+```
+
+To help you keep your fork in sync with the main repo, add an `upstream` remote:
+
+```bash
+$ git remote add upstream git@github.com:multigres/multigres.git
+$ git remote -v
+origin  git@github.com:<yourname>/multigres.git (fetch)
+origin  git@github.com:<yourname>/multigres.git (push)
+upstream        git@github.com:multigres/multigres.git (fetch)
+upstream        git@github.com:multigres/multigres.git (push)
+```
+
+Now to sync your local `main` branch, do this:
+
+```bash
+$ git checkout main
+(main) $ git pull upstream main
+```
+
+Note: In the example output above we prefixed the prompt with `(main)` to stress the fact that the command must be run from the branch `main`.
+
+You can omit the `upstream main` from the `git pull` command when you let your `main` branch always track the main `multigres/multigres` repository. To achieve this, run this command once:
+
+```bash
+(main) $ git branch --set-upstream-to=upstream/main
+```
+
+Now the following command syncs your local `main` branch as well:
+
+```bash
+(main) $ git pull
+```
+
+## Topic Branches
+
+Before you start working on changes, create a topic branch:
+
+```bash
+$ git checkout main
+(main) $ git pull
+(main) $ git checkout -b new-feature
+(new-feature) $ # You are now in the new-feature branch.
+```
+
+As you work in a package, you can run just the unit tests for that package by running `go test` from within that package.
+
+When you're ready to test the whole system, run the full test suite with `make test` from the root of the Git tree.
+If you haven't installed all dependencies for `make test`, you can rely on the CI test results as well.
+These results will be present on your pull request.
+
+## Committing your work
+
+When running `git commit` use the `-s` option to add a Signed-off-by line.
```

Comment:
Somewhere near the commits or PR content here it's probably worth mentioning that we use the conventional commits format for PRs?